planner/memo/implementation.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memo
+
+import (
+	plannercore "github.com/pingcap/tidb/planner/core"
+)
+
+// Implementation defines the interface for cost of physical plan.
+type Implementation interface {
```

Comment:
Move it to `implementation` package?

Reply:
I've tried, this will cause a circular dependency:
1. planner/memo depends on planner/implementation
```go
type Group struct {
	...
    ImplMap map[string]Implementation
	...
}
```
2. planner/implementation depends on planner/memo
```go
type Implementation interface {
    CalcCost(outCount float64, childCosts []float64, children ...*Group) float64
	...
}
```

So I left the interface definition in planner/memo and implementation in planner/implementation